opentelemetry/src/sdk/trace/tracer.rs

```diff
@@ -172,9 +172,14 @@ impl crate::trace::Tracer for Tracer {
         let parent_span_context = builder
             .parent_context
             .as_ref()
-            .or_else(|| Some(cx.span().span_context()).filter(|cx| cx.is_valid()))
-            .or_else(|| cx.remote_span_context())
-            .filter(|cx| cx.is_valid());
+            .or_else(|| {
+                if cx.has_active_span() {
+                    Some(cx.span().span_context())
+                } else {
+                    None
+                }
+            })
+            .or_else(|| cx.remote_span_context());
```

Comment:
We have very similar logic in `NoopTracer`. Do we need to change that, too?

https://github.com/open-telemetry/opentelemetry-rust/blob/bd696420f33295922ddfb62e075e19b6d9fc4487/opentelemetry/src/api/trace/noop.rs#L152-L158

Reply:
Oh yeah good catch